1_prep.R

```diff
@@ -17,36 +17,25 @@ p1 <- list(
              readr::read_csv(p1_lake_cell_tile_xwalk_csv, col_types=cols()) %>%
                filter(site_id %in% p1_nml_site_ids) %>%
                arrange(site_id)),
+  # NetCDF files with munged GCM driver data (one per GCM)
+  # files copied from lake-temperature-model-prep repo
+  tar_target(p1_gcm_ncs, {
+    filename <- sprintf('1_prep/in/GCM_%s.nc', p1_gcm_names)
+    return(filename)
+  }, format = 'file', pattern = map(p1_gcm_names)),
```

Comment:
I don't think you need to map here. It will automatically create multiple filenames and is vectorized. Since this is such a quick target, mapping is probably more overhead than its worth.

Reply:
Actually I need `p1_gcm_ncs` to be a branched target so that I can map over it to build `p1_meteo_feathers`. If I drop the mapping I get this error:
```R
Error: Target p1_meteo_feathers tried to branch over p1_gcm_ncs, which is illegal. Patterns must only branch over explicitly declared targets in the pipeline. 
```

Reply:
Hmmm I know you've told me this before when I have similar feedback but I just don't get why `targets` is requiring you to branch over something that does not need to be mapped. The later step should be able to map over it after-the-fact.

Reply:
hmm now I'm wondering if would map over `p1_gcm_ncs` if it is defined without mapping if I specify `iteration = 'list'`?

Reply:
Nope that didn't work. I agree it is odd. It seems like it might be specific to file targets?